gcloud-java-compute/src/main/java/com/google/gcloud/compute/Snapshot.java

```diff
@@ -0,0 +1,208 @@
+    }
+
+    @Override
+    Builder status(Status status) {
+      infoBuilder.status(status);
+      return this;
+    }
+
+    @Override
+    Builder diskSizeGb(Long diskSizeGb) {
+      infoBuilder.diskSizeGb(diskSizeGb);
+      return this;
+    }
+
+    @Override
+    Builder licenses(List<LicenseId> licenses) {
+      infoBuilder.licenses(licenses);
+      return this;
+    }
+
+    @Override
+    public Builder sourceDisk(DiskId sourceDisk) {
+      infoBuilder.sourceDisk(sourceDisk);
+      return this;
+    }
+
+    @Override
+    Builder sourceDiskId(String sourceDiskId) {
+      infoBuilder.sourceDiskId(sourceDiskId);
+      return this;
+    }
+
+    @Override
+    Builder storageBytes(Long storageBytes) {
+      infoBuilder.storageBytes(storageBytes);
+      return this;
+    }
+
+    @Override
+    Builder storageBytesStatus(StorageBytesStatus storageBytesStatus) {
+      infoBuilder.storageBytesStatus(storageBytesStatus);
+      return this;
+    }
+
+    @Override
+    public Snapshot build() {
+      return new Snapshot(compute, infoBuilder);
+    }
+  }
+
+  Snapshot(Compute compute, SnapshotInfo.BuilderImpl infoBuilder) {
+    super(infoBuilder);
+    this.compute = checkNotNull(compute);
+    this.options = compute.options();
+  }
+
+  /**
+   * Checks if this snapshot exists.
+   *
+   * @return {@code true} if this snapshot exists, {@code false} otherwise
+   * @throws ComputeException upon failure
+   */
+  public boolean exists() {
+    return reload(Compute.SnapshotOption.fields()) != null;
+  }
+
+  /**
+   * Fetches current snapshot's latest information. Returns {@code null} if the snapshot does not
+   * exist.
+   *
+   * @param options snapshot options
+   * @return a {@code Snapshot} object with latest information or {@code null} if not found
+   * @throws ComputeException upon failure
+   */
+  public Snapshot reload(Compute.SnapshotOption... options) {
+    return compute.getSnapshot(snapshotId().snapshot(), options);
+  }
+
+  /**
+   * Deletes this snapshot.
+   *
+   * @return a global operation if delete request was successfully sent, {@code null} if the
+   *     snapshot was not found
+   * @throws ComputeException upon failure
+   */
+  public Operation delete(Compute.OperationOption... options) {
+    return compute.deleteSnapshot(snapshotId(), options);
+  }
+
+  /**
+   * Returns the snapshot's {@code Compute} object used to issue requests.
+   */
+  public Compute compute() {
+    return compute;
+  }
+
+  @Override
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
+
+  @Override
+  public final boolean equals(Object obj) {
+    return obj instanceof Snapshot
+        && Objects.equals(toPb(), ((Snapshot) obj).toPb())
+        && Objects.equals(options, ((Snapshot) obj).options);
+  }
+
+  @Override
+  public final int hashCode() {
+    return Objects.hash(super.hashCode(), options);
+  }
+
+  private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundException {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid prefixing parameters by in, out or inOut. Uses Javadoc to document this behavior.](https://www.codacy.com/app/mziccard/gcloud-java/file/2301651869/issues/source?bid=3140968&fileBranchId=3194371#l199)